artiq/firmware/libboard_misoc/lib.rs

```diff
@@ -43,3 +43,5 @@ pub mod io_expander;
 pub mod net_settings;
 #[cfg(has_slave_fpga_cfg)]
 pub mod slave_fpga;
+#[cfg(any(soc_platform = "kasli", soc_platform = "metlino"))]
+pub mod flash;
```

Comment:
This module shouldn't be called flash if all it does is talk to the ICAP.
Do we need a module for just that function?

Reply:
Because later I'll put the flash functions into this file. Should I put it into `or1k/boot.rs`?

Reply:
It's not OR1K-specific. ``flash`` is also a bad name because there's already ``spiflash``.

Reply:
Then, I think it's better to keep this file at this place. How about renaming it as `flash_tools`?

Reply:
Why not simply add to spiflash?